patch/cisco-acpi-spi-nor.patch

```diff
@@ -0,0 +1,46 @@
+From 68853e9c3d5e19a0145cb75226d239e8a869742d Mon Sep 17 00:00:00 2001
+From: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+Date: Mon, 19 Jul 2021 13:33:44 -0700
+Subject: [PATCH] mtd: m25p80 spi driver update to support ACPI table match
+
+Current m25p80 spi nor driver does not support ACPI table match.
+Configuration done through ACPI tables is not recognized by driver.
+Cisco 8000 platform configures NOR flash partition information
+through ACPI table which is not recognized by m25p80.
+
+Added support in m25p80 to perform ACPI table match.
+
+Signed-off-by: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+---
+ drivers/mtd/devices/m25p80.c | 10 ++++++++++
+ 1 file changed, 10 insertions(+)
+
+diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
+index c4a1d04b8..7a291b649 100644
+--- a/drivers/mtd/devices/m25p80.c
++++ b/drivers/mtd/devices/m25p80.c
+@@ -319,11 +319,21 @@ static const struct of_device_id m25p_of_table[] = {
+ };
+ MODULE_DEVICE_TABLE(of, m25p_of_table);
+ 
++#ifdef CONFIG_ACPI
++static const struct acpi_device_id m25p_acpi_table[] = {
++	{ "JEDEC,SPI-NOR", 0 },
++	{ "ACPI0000", 0 },
++	{}
++};
++MODULE_DEVICE_TABLE(acpi, m25p_acpi_table);
++#endif
++
+ static struct spi_mem_driver m25p80_driver = {
+ 	.spidrv = {
+ 		.driver = {
+ 			.name	= "m25p80",
+ 			.of_match_table = m25p_of_table,
++			.acpi_match_table = m25p_acpi_table,
```

Comment:
The examples and code in the Linux kernel use `ACPI_PTR(m25p_acpi_table)`.

    $ git grep -C2 acpi_match_table

Why don’t you?

Reply:
> Thank you for improving the patch. Have you sent it upstream for review as talked about last week?

Hi Paul,

I haven't sent this patch to upstream yet. First stage we are focusing on getting changes integrated to sonic-linux-kernel. Second stage will be to send patches upstream for review.

Reply:
> ACPI_PTR(m25p_acpi_table)

Addressed as per review comments.